test/e2e/azure_test.go

```diff
@@ -122,6 +122,17 @@ var _ = Describe("Workoad cluster creation", func() {
 				WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
 				WaitForMachineDeployments:    e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
 			})
+
+			Context("in a cluster with one or more nodes", func() {
+				AzureLBSpec(ctx, func() AzureLBSpecInput {
```

Comment:
I don't think a test would run with an `It`

see here https://onsi.github.io/ginkgo/

Reply:
The problem is that we're taking advantage of the workload cluster already being created to run some additional test assertions. So we can't wrap the `AzureLBSpec` in an `It` because they can't be nested (as far as I can tell), and if they're at the same level we can't guarantee the cluster creation ran before the LB test.

Maybe a better way to organize this would be to put the cluster creation in a `BeforeSuite` and then have `It`s that test functionality against the running cluster? That seemed like too much of a refactor to do here.

Reply:
Also, as far as I can tell the new test spec is being run as intended.

Reply:
I think that standard way i've seen before is using a `By` step

Reply:
I guess I'm misunderstanding. The `AzureLBSpec` consists only of `By` steps and does seem to run successfully:

```
INFO: Waiting for the worker machines to be provisioned
[1mSTEP[0m: waiting for the workload nodes to exist
[1mSTEP[0m: creating a Kubernetes client to the workload cluster
[1mSTEP[0m: creating an nginx deployment
[1mSTEP[0m: waiting for deployment default/ingress-nginx to be available
[1mSTEP[0m: creating an internal Load Balancer service
[1mSTEP[0m: waiting for service default/ingress-nginx-ilb to be available
[1mSTEP[0m: connecting to the internal LB service from a curl pod
[1mSTEP[0m: waiting for job default/curl-to-ilb-job to be complete
[1mSTEP[0m: creating an external Load Balancer service
[1mSTEP[0m: waiting for service default/ingress-nginx-elb to be available
[1mSTEP[0m: connecting to the external LB service from a curl pod
[1mSTEP[0m: waiting for job default/curl-to-elb-job to be complete
[1mSTEP[0m: connecting directly to the external LB service
[1mSTEP[0m: deleting the test resources
[1mSTEP[0m: Dumping all the Cluster API resources in the "create-workload-cluster-3vb4me" namespace
```

Should I replace the `Context` with a `By`? Sorry, I'm probably missing some ginkgo pattern. I had to get creative because as I said above, I wanted to re-use the existing cluster and that's already inside an `It` clause.

Reply:
if its working as expected, you don't have to change anything